projects/plugins/crm/src/automation/data-types/class-data-type-contact.php

```diff
@@ -0,0 +1,113 @@
+<?php
+/**
+ * Contact Data Type.
+ *
+ * @package automattic/jetpack-crm
+ * @since $$next-version$$
+ */
+
+namespace Automattic\Jetpack\CRM\Automation\Data_Types;
+
+/**
+ * Contact Data Type
+ *
+ * @since $$next-version$$
+ */
+class Data_Type_Contact extends Data_Type_Base {
+
+	/**
+	 * {@inheritDoc}
+	 */
+	public static function get_slug(): string {
+		return 'contact';
+	}
+
+	/**
+	 * Constructor.
+	 *
+	 * We process the entity data before passing it to validation.
+	 * You can learn more in the "unify_data" method.
+	 *
+	 * @see self::unify_data()
+	 *
+	 * @param mixed $entity The contact entity data.
+	 * @return void
+	 *
+	 * @throws \Automattic\Jetpack\CRM\Automation\Data_Type_Exception If the entity is not valid.
+	 */
+	public function __construct( $entity ) {
+		$entity = $this->unify_data( $entity );
+
+		parent::__construct( $entity );
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	public function get_id() {
+		return $this->entity['id'];
+	}
+
+	/**
+	 * Validate the contact entity data.
+	 *
+	 * @since $$next-version$$
+	 *
+	 * @param mixed $entity Contact entity data to validate.
+	 * @return bool Whether the entity is valid or not.
+	 */
+	public function validate_entity( $entity ): bool {
+		// We do not have a model/object class for contacts, so as the very
+		// minimum we need to make sure we received an array since that's
+		// the response the CRM DAL returns.
+		if ( ! is_array( $entity ) ) {
+			return false;
+		}
+
+		// We could look for even more fields, but this should ensure we have
+		// received a data array that looks valid enough.
+		$required_fields = array(
+			'id',
+			'email',
+			'status',
+			'fname',
+			'lname',
+		);
+
+		foreach ( $required_fields as $field ) {
+			if ( ! isset( $entity[ $field ] ) ) {
+				return false;
+			}
+		}
+
+		return true;
+	}
+
+	/**
+	 * Unify how CRM contact data is formatted.
+	 *
+	 * zbsDAL_contacts::getContact() formats the data when it returns results
+	 * but some hooks like "contact.new" passes an older, raw, formatting which
+	 * means we have two different types of formats being passed around.
+	 * This method attempts to unify the formatting, so we only have to work
+	 * with a single version of the formatting (the most recent one).
+	 *
+	 * @since $$next-version$$
+	 *
+	 * @param mixed $entity The data we want to potentially prepare.
+	 * @return array The unified data.
+	 */
+	public function unify_data( $entity ): array {
+		if ( ! is_array( $entity ) ) {
+			return $entity;
```

Comment:
If it's not an array it cannot return an array

Reply:
Fixed here: [176656a51f6dd3574cb91a9d01ca9289611b096b](https://github.com/Automattic/jetpack/pull/32167/commits/176656a51f6dd3574cb91a9d01ca9289611b096b)